src/index.js

```diff
@@ -0,0 +1,286 @@
+    ) {
+      return [];
+    }
+
+    remoteContext.experiments.allocations.forEach(function(alloc) {
+      eventBeacon.emit('confirmation', assign({
+        uid: alloc.uid,
+        sid: alloc.sid,
+        eid: alloc.eid,
+        cid: alloc.cid
+      }, context.remoteContext));
+    });
+    eventBeacon.flush();
+    emit(context, EvolvClient.CONFIRMED);
+  };
+
+  /**
+   * Marks a consumer as unsuccessfully retrieving and / or applying requested values, making them ineligible for
+   * inclusion in optimization statistics.
+   */
+  this.contaminate = function() {
+    const remoteContext = context.remoteContext;
+    if (
+      !remoteContext.experiments ||
+      !remoteContext.experiments.allocations || !remoteContext.experiments.allocations.length
+    ) {
+      return [];
+    }
+
+    remoteContext.experiments.allocations.forEach(function(alloc) {
+      eventBeacon.emit('contamination', assign({
+        uid: alloc.uid,
+        sid: alloc.sid,
+        eid: alloc.eid,
+        cid: alloc.cid
+      }, context.remoteContext));
+    });
+    eventBeacon.flush();
+    emit(context, EvolvClient.CONTAMINATED);
+  };
+
+  /**
+   * Initializes the client with required context information.
+   *
+   * @param {String} uid A globally unique identifier for the current participant.
+   * @param {String} sid A globally unique session identifier for the current participant.
+   * @param {Object} remoteContext A map of data used for evaluating context predicates and analytics.
+   * @param {Object} localContext A map of data used only for evaluating context predicates.
+   */
+  this.initialize = function (uid, sid, remoteContext, localContext) {
+    if (initialized) {
+      throw Error('Evolv: Client is already initialized');
+    }
+
+    if (!uid) {
+      throw new Error('Evolv: "uid" must be specified');
+    }
+
+    if (!sid) {
+      throw new Error('Evolv: "sid" must be specified');
+    }
+
+    context.initialize(uid, sid, remoteContext, localContext);
+    store.initialize(context);
+
+    waitFor(context, CONTEXT_INITIALIZED, function(type, ctx) {
+      contextBeacon.emit(type, context.remoteContext);
+    });
+    waitFor(context, CONTEXT_VALUE_ADDED, function(type, key, value, local) {
+      if (local) {
+        return;
+      }
+
+      contextBeacon.emit(type, { key: key, value: value });
+    });
+    waitFor(context, CONTEXT_VALUE_CHANGED, function(type, key, value, before, local) {
+      if (local) {
+        return;
+      }
+
+      contextBeacon.emit(type, { key: key, value: value });
+    });
+    waitFor(context, CONTEXT_VALUE_REMOVED, function(type, key, local) {
+      if (local) {
+        return;
+      }
+
+      contextBeacon.emit(type, { key: key });
+    });
+
+    if (options.autoConfirm) {
+      waitFor(EFFECTIVE_GENOME_UPDATED, this.confirm.bind(this));
+      waitFor(REQUEST_FAILED, this.contaminate.bind(this));
```

Comment:
Should these usages of `waitFor()` not take 3 arguments? `waitFor()` does not appear to handle variadic arguments.